### ISSUE TITLE
STS: assumeRoleWithWebIdentity doesn't work via edge port

### ISSUE THREAD

**Issue by nhoughto** [localstack/localstack#3190]
# Type of request: This is a ...

[x] bug report
[ ] feature request

# Detailed description
The recent change of only allowing traffic via edge port got us to refactor our STS impl, and we have found that STS `assumeRoleWithWebIdentity` doesn't work via edge port, it does via the dedicated STS port. This is a blocker to upgrade as the dedicated port are removed now I believe? normal `assume-role` works its just web identity..
...

## Expected behavior

assumeRoleWithWebIdentity works

...

## Actual behavior

It responses as if its called S3, log is below:
```
b'<ListAllMyBucketsResult xmlns="http://s3.amazonaws.com/doc/2006-03-01"><Owner><ID>bcaf1ffd86f41161ca5fb16fd081034f</ID><DisplayName>webfile</DisplayName></Owner><Buckets><Bucket><Name>pro1-npe-tfstate</Name><CreationDate>2020-10-28T06:18:15.000Z</CreationDate></Bucket></Buckets></ListAllMyBucketsResult>'
```

...

# Steps to reproduce

## Command used to start LocalStack
Latest 0.11.6 localstack docker with `ses,sqs,sns,sts,lambda,iam,dynamodb,s3` services

...

## Client code (AWS SDK code snippet, or sequence of "awslocal" commands)
`awslocal sts assume-role-with-web-identity --web-identity-token token --role-arn 123jk1j221ljk1lk12038102981blah --role-session-name blah --region us-east-1 --debug`

log
```
bash-5.0# awslocal sts assume-role-with-web-identity --web-identity-token token --role-arn 123jk1j221ljk1lk12038102981blah --role-session-name blah --region us-east-1 --debug
2020-10-28 06:34:04,198 - MainThread - awscli.clidriver - DEBUG - CLI version: aws-cli/1.18.152 Python/3.8.2 Linux/4.19.76-linuxkit botocore/1.18.11
2020-10-28 06:34:04,198 - MainThread - awscli.clidriver - DEBUG - Arguments entered to CLI: ['--endpoint-url=http://localhost:4566', 'sts', 'assume-role-with-web-identity', '--web-identity-token', 'token', '--role-arn', '123jk1j221ljk1lk12038102981blah', '--role-session-name', 'blah', '--region', 'us-east-1', '--debug']
2020-10-28 06:34:04,199 - MainThread - botocore.hooks - DEBUG - Event session-initialized: calling handler <function add_scalar_parsers at 0x7fc336762b80>
2020-10-28 06:34:04,200 - MainThread - botocore.hooks - DEBUG - Event session-initialized: calling handler <function register_uri_param_handler at 0x7fc336c08af0>
2020-10-28 06:34:04,201 - MainThread - botocore.hooks - DEBUG - Event session-initialized: calling handler <function inject_assume_role_provider_cache at 0x7fc336a05280>
2020-10-28 06:34:04,205 - MainThread - botocore.hooks - DEBUG - Event session-initialized: calling handler <function attach_history_handler at 0x7fc3368a4670>
2020-10-28 06:34:04,206 - MainThread - botocore.loaders - DEBUG - Loading JSON file: /opt/code/localstack/.venv/lib/python3.8/site-packages/botocore/data/sts/2011-06-15/service-2.json
2020-10-28 06:34:04,208 - MainThread - botocore.hooks - DEBUG - Event building-command-table.sts: calling handler <function add_waiters at 0x7fc336770700>
2020-10-28 06:34:04,218 - MainThread - awscli.clidriver - DEBUG - OrderedDict([('role-arn', <awscli.arguments.CLIArgument object at 0x7fc3366740d0>), ('role-session-name', <awscli.arguments.CLIArgument object at 0x7fc336674100>), ('web-identity-token', <awscli.arguments.CLIArgument object at 0x7fc336674130>), ('provider-id', <awscli.arguments.CLIArgument object at 0x7fc336674160>), ('policy-arns', <awscli.arguments.ListArgument object at 0x7fc336674190>), ('policy', <awscli.arguments.CLIArgument object at 0x7fc3366741c0>), ('duration-seconds', <awscli.arguments.CLIArgument object at 0x7fc3366741f0>)])
2020-10-28 06:34:04,218 - MainThread - botocore.hooks - DEBUG - Event building-argument-table.sts.assume-role-with-web-identity: calling handler <function add_streaming_output_arg at 0x7fc336767280>
2020-10-28 06:34:04,219 - MainThread - botocore.hooks - DEBUG - Event building-argument-table.sts.assume-role-with-web-identity: calling handler <function add_cli_input_json at 0x7fc336a05d30>
2020-10-28 06:34:04,220 - MainThread - botocore.hooks - DEBUG - Event building-argument-table.sts.assume-role-with-web-identity: calling handler <function unify_paging_params at 0x7fc3367ecd30>
2020-10-28 06:34:04,229 - MainThread - botocore.loaders - DEBUG - Loading JSON file: /opt/code/localstack/.venv/lib/python3.8/site-packages/botocore/data/sts/2011-06-15/paginators-1.json
2020-10-28 06:34:04,230 - MainThread - botocore.hooks - DEBUG - Event building-argument-table.sts.assume-role-with-web-identity: calling handler <function add_generate_skeleton at 0x7fc33684c790>
2020-10-28 06:34:04,231 - MainThread - botocore.hooks - DEBUG - Event before-building-argument-table-parser.sts.assume-role-with-web-identity: calling handler <bound method OverrideRequiredArgsArgument.override_required_args of <awscli.customizations.cliinputjson.CliInputJSONArgument object at 0x7fc3366743a0>>
2020-10-28 06:34:04,231 - MainThread - botocore.hooks - DEBUG - Event before-building-argument-table-parser.sts.assume-role-with-web-identity: calling handler <bound method GenerateCliSkeletonArgument.override_required_args of <awscli.customizations.generatecliskeleton.GenerateCliSkeletonArgument object at 0x7fc336674430>>
2020-10-28 06:34:04,233 - MainThread - botocore.hooks - DEBUG - Event load-cli-arg.sts.assume-role-with-web-identity.role-arn: calling handler <awscli.paramfile.URIArgumentHandler object at 0x7fc33669bca0>
2020-10-28 06:34:04,235 - MainThread - botocore.hooks - DEBUG - Event process-cli-arg.sts.assume-role-with-web-identity: calling handler <awscli.argprocess.ParamShorthandParser object at 0x7fc3369d9370>
2020-10-28 06:34:04,235 - MainThread - awscli.arguments - DEBUG - Unpacked value of '123jk1j221ljk1lk12038102981blah' for parameter "role_arn": '123jk1j221ljk1lk12038102981blah'
2020-10-28 06:34:04,236 - MainThread - botocore.hooks - DEBUG - Event load-cli-arg.sts.assume-role-with-web-identity.role-session-name: calling handler <awscli.paramfile.URIArgumentHandler object at 0x7fc33669bca0>
2020-10-28 06:34:04,236 - MainThread - botocore.hooks - DEBUG - Event process-cli-arg.sts.assume-role-with-web-identity: calling handler <awscli.argprocess.ParamShorthandParser object at 0x7fc3369d9370>
2020-10-28 06:34:04,237 - MainThread - awscli.arguments - DEBUG - Unpacked value of 'blah' for parameter "role_session_name": 'blah'
2020-10-28 06:34:04,238 - MainThread - botocore.hooks - DEBUG - Event load-cli-arg.sts.assume-role-with-web-identity.web-identity-token: calling handler <awscli.paramfile.URIArgumentHandler object at 0x7fc33669bca0>
2020-10-28 06:34:04,239 - MainThread - botocore.hooks - DEBUG - Event process-cli-arg.sts.assume-role-with-web-identity: calling handler <awscli.argprocess.ParamShorthandParser object at 0x7fc3369d9370>
2020-10-28 06:34:04,239 - MainThread - awscli.arguments - DEBUG - Unpacked value of 'token' for parameter "web_identity_token": 'token'
2020-10-28 06:34:04,240 - MainThread - botocore.hooks - DEBUG - Event load-cli-arg.sts.assume-role-with-web-identity.provider-id: calling handler <awscli.paramfile.URIArgumentHandler object at 0x7fc33669bca0>
2020-10-28 06:34:04,241 - MainThread - botocore.hooks - DEBUG - Event load-cli-arg.sts.assume-role-with-web-identity.policy-arns: calling handler <awscli.paramfile.URIArgumentHandler object at 0x7fc33669bca0>
2020-10-28 06:34:04,241 - MainThread - botocore.hooks - DEBUG - Event load-cli-arg.sts.assume-role-with-web-identity.policy: calling handler <awscli.paramfile.URIArgumentHandler object at 0x7fc33669bca0>
2020-10-28 06:34:04,242 - MainThread - botocore.hooks - DEBUG - Event load-cli-arg.sts.assume-role-with-web-identity.duration-seconds: calling handler <awscli.paramfile.URIArgumentHandler object at 0x7fc33669bca0>
2020-10-28 06:34:04,243 - MainThread - botocore.hooks - DEBUG - Event load-cli-arg.sts.assume-role-with-web-identity.cli-input-json: calling handler <awscli.paramfile.URIArgumentHandler object at 0x7fc33669bca0>
2020-10-28 06:34:04,243 - MainThread - botocore.hooks - DEBUG - Event load-cli-arg.sts.assume-role-with-web-identity.generate-cli-skeleton: calling handler <awscli.paramfile.URIArgumentHandler object at 0x7fc33669bca0>
2020-10-28 06:34:04,243 - MainThread - botocore.hooks - DEBUG - Event calling-command.sts.assume-role-with-web-identity: calling handler <bound method CliInputJSONArgument.add_to_call_parameters of <awscli.customizations.cliinputjson.CliInputJSONArgument object at 0x7fc3366743a0>>
2020-10-28 06:34:04,243 - MainThread - botocore.hooks - DEBUG - Event calling-command.sts.assume-role-with-web-identity: calling handler <bound method GenerateCliSkeletonArgument.generate_json_skeleton of <awscli.customizations.generatecliskeleton.GenerateCliSkeletonArgument object at 0x7fc336674430>>
2020-10-28 06:34:04,244 - MainThread - botocore.credentials - DEBUG - Looking for credentials via: env
2020-10-28 06:34:04,245 - MainThread - botocore.credentials - INFO - Found credentials in environment variables.
2020-10-28 06:34:04,246 - MainThread - botocore.loaders - DEBUG - Loading JSON file: /opt/code/localstack/.venv/lib/python3.8/site-packages/botocore/data/endpoints.json
2020-10-28 06:34:04,253 - MainThread - botocore.hooks - DEBUG - Event choose-service-name: calling handler <function handle_service_name_alias at 0x7fc3371b4040>
2020-10-28 06:34:04,255 - MainThread - botocore.hooks - DEBUG - Event creating-client-class.sts: calling handler <function add_generate_presigned_url at 0x7fc3372544c0>
2020-10-28 06:34:04,262 - MainThread - botocore.endpoint - DEBUG - Setting sts timeout as (60, 60)
2020-10-28 06:34:04,263 - MainThread - botocore.loaders - DEBUG - Loading JSON file: /opt/code/localstack/.venv/lib/python3.8/site-packages/botocore/data/_retry.json
2020-10-28 06:34:04,264 - MainThread - botocore.client - DEBUG - Registering retry handlers for service: sts
2020-10-28 06:34:04,266 - MainThread - botocore.hooks - DEBUG - Event before-parameter-build.sts.AssumeRoleWithWebIdentity: calling handler <function generate_idempotent_uuid at 0x7fc3371d40d0>
2020-10-28 06:34:04,266 - MainThread - botocore.hooks - DEBUG - Event before-call.sts.AssumeRoleWithWebIdentity: calling handler <function inject_api_version_header_if_needed at 0x7fc3371d5940>
2020-10-28 06:34:04,267 - MainThread - botocore.endpoint - DEBUG - Making request for OperationModel(name=AssumeRoleWithWebIdentity) with params: {'url_path': '/', 'query_string': '', 'method': 'POST', 'headers': {'Content-Type': 'application/x-www-form-urlencoded; charset=utf-8', 'User-Agent': 'aws-cli/1.18.152 Python/3.8.2 Linux/4.19.76-linuxkit botocore/1.18.11'}, 'body': {'Action': 'AssumeRoleWithWebIdentity', 'Version': '2011-06-15', 'RoleArn': '123jk1j221ljk1lk12038102981blah', 'RoleSessionName': 'blah', 'WebIdentityToken': 'token'}, 'url': 'http://localhost:4566/', 'context': {'client_region': 'us-east-1', 'client_config': <botocore.config.Config object at 0x7fc336674f70>, 'has_streaming_input': False, 'auth_type': None}}
2020-10-28 06:34:04,267 - MainThread - botocore.hooks - DEBUG - Event request-created.sts.AssumeRoleWithWebIdentity: calling handler <bound method RequestSigner.handler of <botocore.signers.RequestSigner object at 0x7fc336674d90>>
2020-10-28 06:34:04,267 - MainThread - botocore.hooks - DEBUG - Event choose-signer.sts.AssumeRoleWithWebIdentity: calling handler <function disable_signing at 0x7fc3371d4550>
2020-10-28 06:34:04,268 - MainThread - botocore.endpoint - DEBUG - Sending http request: <AWSPreparedRequest stream_output=False, method=POST, url=http://localhost:4566/, headers={'Content-Type': b'application/x-www-form-urlencoded; charset=utf-8', 'User-Agent': b'aws-cli/1.18.152 Python/3.8.2 Linux/4.19.76-linuxkit botocore/1.18.11', 'Content-Length': '135'}>
2020-10-28 06:34:04,270 - MainThread - urllib3.connectionpool - DEBUG - Starting new HTTP connection (1): localhost:4566
2020-10-28 06:34:04,284 - MainThread - urllib3.connectionpool - DEBUG - http://localhost:4566 "POST / HTTP/1.1" 200 304
2020-10-28 06:34:04,285 - MainThread - botocore.parsers - DEBUG - Response headers: {'content-type': 'application/xml; charset=utf-8', 'content-length': '304', 'access-control-allow-origin': '*', 'last-modified': 'Wed, 28 Oct 2020 06:34:04 GMT', 'x-amz-request-id': 'DB9FA7AECA054A18', 'x-amz-id-2': 'MzRISOwyjmnupDB9FA7AECA054A187/JypPGXLh0OVFGcJaaO3KW/hRAqKOpIEEp', 'access-control-allow-methods': 'HEAD,GET,PUT,POST,DELETE,OPTIONS,PATCH', 'access-control-allow-headers': 'authorization,content-type,content-md5,cache-control,x-amz-content-sha256,x-amz-date,x-amz-security-token,x-amz-user-agent,x-amz-target,x-amz-acl,x-amz-version-id,x-localstack-target,x-amz-tagging', 'access-control-expose-headers': 'x-amz-version-id', 'connection': 'close', 'date': 'Wed, 28 Oct 2020 06:34:04 GMT', 'server': 'hypercorn-h11'}
2020-10-28 06:34:04,285 - MainThread - botocore.parsers - DEBUG - Response body:
b'<ListAllMyBucketsResult xmlns="http://s3.amazonaws.com/doc/2006-03-01"><Owner><ID>bcaf1ffd86f41161ca5fb16fd081034f</ID><DisplayName>webfile</DisplayName></Owner><Buckets><Bucket><Name>pro1-npe-tfstate</Name><CreationDate>2020-10-28T06:18:15.000Z</CreationDate></Bucket></Buckets></ListAllMyBucketsResult>'
2020-10-28 06:34:04,287 - MainThread - awscli.clidriver - DEBUG - Exception caught in main()
Traceback (most recent call last):
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/awscli/clidriver.py", line 217, in main
    return command_table[parsed_args.command](remaining, parsed_args)
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/awscli/clidriver.py", line 358, in __call__
    return command_table[parsed_args.operation](remaining, parsed_globals)
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/awscli/clidriver.py", line 527, in __call__
    return self._operation_caller.invoke(
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/awscli/clidriver.py", line 649, in invoke
    response = self._make_client_call(
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/awscli/clidriver.py", line 661, in _make_client_call
    response = getattr(client, xform_name(operation_name))(
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/botocore/client.py", line 357, in _api_call
    return self._make_api_call(operation_name, kwargs)
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/botocore/client.py", line 662, in _make_api_call
    http, parsed_response = self._make_request(
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/botocore/client.py", line 682, in _make_request
    return self._endpoint.make_request(operation_model, request_dict)
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/botocore/endpoint.py", line 102, in make_request
    return self._send_request(request_dict, operation_model)
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/botocore/endpoint.py", line 134, in _send_request
    success_response, exception = self._get_response(
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/botocore/endpoint.py", line 166, in _get_response
    success_response, exception = self._do_get_response(
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/botocore/endpoint.py", line 217, in _do_get_response
    parsed_response = parser.parse(
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/botocore/parsers.py", line 245, in parse
    parsed = self._do_parse(response, shape)
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/botocore/parsers.py", line 523, in _do_parse
    return self._parse_body_as_xml(response, shape, inject_metadata=True)
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/botocore/parsers.py", line 532, in _parse_body_as_xml
    start = self._find_result_wrapped_shape(
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/botocore/parsers.py", line 542, in _find_result_wrapped_shape
    return mapping[element_name]
KeyError: 'AssumeRoleWithWebIdentityResult'
2020-10-28 06:34:04,288 - MainThread - awscli.clidriver - DEBUG - Exiting with rc 255

'AssumeRoleWithWebIdentityResult'
bash-5.0# 

```

**Comment by nhoughto** [comment]
workaround for this until fix

```
sed -i "s|data_bytes = to_bytes(data or '')|data_bytes = to_bytes(data or '')\n    if b'Action=AssumeRoleWithWebIdentity' in data_bytes or 'Action=AssumeRoleWithWebIdentity' in path:\n        return 'sts', config.PORT_STS|g" ./localstack/services/edge.py
```

**Comment by whummer** [comment]
Thanks for reporting @nhoughto . This should be fixed in #3227 - can you please pull the `latest` Docker image and give it another try? Please report here if the problem persists. Thanks
